pcdsdevices/epics_motor.py

```diff
@@ -89,12 +95,20 @@ class EpicsMotorInterface(FltMvInterface, EpicsMotor):
     EpicsMotor.low_limit_travel.kind = Kind.config
     EpicsMotor.direction_of_travel.kind = Kind.normal
 
+    # Registry of log filters by device name
+    _motion_error_log_filters: dict[str, filter] = {}
+    _moved_in_session: bool
     _egu: str
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._egu = ''
         self.motor_egu.subscribe(self._cache_egu)
+        self._install_motion_error_filter()
```

Comment:
I think the hard reference in`_motion_error_log_filters` will stop `EpicsMotorInterface` instances from being garbage collected. Not a big deal in production where our objects aren't created for temporary usage, but... may affect the test suite a bit, perhaps? 🤷‍♂️

Reply:
I think you're correct, I need to use the weakref stuff

Reply:
Hmm, it turns out that a weakref to a method is always immediately dropped, or at least it is in this context. I either don't need to do this or I need to switch it to a weakref mapping of objects instead of callables.

Reply:
If we do need a weakref mapping, I'm just going to add an ophyd instantiation callback during the pcdsdevices startup

Reply:
https://docs.python.org/3/library/weakref.html#weakref.WeakMethod

Reply:
Heading down this WeakMethod path is starting to make the code more complicated than it needs to be for a log filter. I think I'm going to back up and use the ophyd registry stuff because I think it will make everything cleaner (plus we may be interested in the registry itself for other cases)